types/index.d.ts

```diff
@@ -3364,9 +3372,23 @@ declare namespace math {
      * Default value is 10000.
      */
     fractionsLimit?: number;
+    /** A boolean which is `false` by default. */
+    consoleDebug?: boolean;
+    /**
+     * gives properties of each operator, which determine what simplifications
+     * are allowed. Properties are commutative, associative, total (whether
+     * the operation is defined for all arguments), and trivial (whether
+     * the operation applied to a single argument leaves that argument 
+     * unchanged).
+     */
+    context?: SimplifyContext
   }
 
-  type SimplifyRule = { l: string; r: string } | string | ((node: MathNode) => MathNode);
+  type SimplifyRule = 
+  { l: string; r: string; repeat?: boolean; assuming?: SimplifyContext; imposeContext?: SimplifyContext } 
+  | { s: string; repeat?: boolean; assuming?: SimplifyContext; imposeContext?: SimplifyContext } 
```

Comment:
Wasn't sure if `SimplifyContext` was the best name for the `assuming` type, but wasn't sure it was worth creating a new type called something like `SimplifyAssumptions` instead, if it's going to be the same as `SimplifyContext`. What do you think?